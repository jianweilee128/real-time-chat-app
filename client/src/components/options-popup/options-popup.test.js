import React from "react";
import { shallow } from "enzyme";
import OptionsPopup from "./options-popup.component";

describe("OptionsPopup component", () => {
  let wrapper;
  let mockSetToggleOptionsPopup;

  beforeEach(() => {
    mockSetToggleOptionsPopup = jest.fn();

    const mockProps = {
      user: { name: "JohnDoe", id: "123" },
      currentRoom: [("mockRoom", "123")],
      setToggleOptionsPopup: mockSetToggleOptionsPopup,
    };

    wrapper = shallow(<OptionsPopup.WrappedComponent {...mockProps} />);
  });

  it("should render OptionsPopup component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setToggleOptionsPopup function when clicked", () => {
    wrapper.find(".close-icon").simulate("click");
    expect(mockSetToggleOptionsPopup).toHaveBeenCalled();
  });
});
