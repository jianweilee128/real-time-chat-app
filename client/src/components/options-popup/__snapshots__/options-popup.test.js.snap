// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsPopup component should render OptionsPopup component 1`] = `ShallowWrapper {}`;
